src/herder/test/HerderTests.cpp

```diff
@@ -3801,22 +3752,28 @@ herderExternalizesValuesWithProtocol(uint32_t version)
     REQUIRE(currentALedger() >= currentLedger);
     REQUIRE(currentCLedger() == currentLedger);
 
+    // Arm the upgrade, but don't close the upgrade ledger yet
```

Comment:
Yeah, I should clarify that this doesn't close the upgrade ledger, but does close ledger in order to set up the upgrade.

That being said, I think this test does the same thing the previous test did? In particular, the `setupUpgradeAtNextLedger` function previously commited the upgrade directly to the DB, then called `getHerder().setUpgrades()` on the commited upgrade set. The test then disconnected C and closed a few ledgers., then reconnected C.

This does the same thing, except it uses load gen to generate the `ConfigUpgradeSet` instead of directly comiting. This is fine though, as we set the upgrade, but don't close the upgrade ledger until after we disconnect C.

Reply:
> This is fine though, as we set the upgrade, but don't close the upgrade ledger until after we disconnect C.

But that's not what `upgradeSorobanNetworkConfig` does, is it? `upgradeSorobanNetworkConfig` waits for the upgrade to be applied after arming it.

Reply:
No, I added the `applyUpgrade` flag to determine if `upgradeSorobanNetworkConfig` closes ledgers after arming the upgrade. In this test it's set to false so `upgradeSorobanNetworkConfig` and arms the upgrade, then returns before closing any additional ledgers.

Reply:
Ah, sorry, I've been looking at the current implementation.